src/meta/meta_ckpt.c

```diff
@@ -736,6 +736,36 @@ __wt_metadata_update_base_write_gen(WT_SESSION_IMPL *session, const char *config
     return (0);
 }
 
+/*
+ * __wt_metadata_update_all_base_write_gen --
```

Comment:
Is there a way to refactor this to avoid adding another external call to update the `base_write_gen`? This will now create 3 different, but very similar, ways to enter this code and update that value. Knowing which one is the correct one to call in the future will be confusing. Is it possible to combine and/or add a boolean to avoid that?

Reply:
I didn't any better to combine those functions. I have renamed the function names and their description so that it shouldn't cause any problem to use them later. Can you please review the renamed functions?